### ISSUE TITLE
`no-relative-imports` seems broken

### ISSUE THREAD

**Issue by jeffijoe** [Microsoft/tslint-microsoft-contrib#168]
Refer to the following screenshot:

![image](https://cloud.githubusercontent.com/assets/2505474/16584775/e76e3014-42bd-11e6-9b68-016da55877f3.png)

```ts
import { Author } from './Author';
```

This violates the rule, even though `Author` is a local file, and not an external module.

**Comment by HamletDRC** [comment]
Hi @jeffijoe 

The violation you see is indeed working correctly. The rule would like you to rewrite your import like this: 

    import { Author } from 'Author';

The advantages are: 
* searching for "from 'Author'" will now return every single import of Author regardless of which subdirectory Post.ts is sitting in
* You can move Post.ts to any subdirectory you want and you will not have to change all your imports
* You can copy/paste a set of imports from one file to another without having to change all the paths

I will update the readme.md file with a little more info about this.

**Comment by jeffijoe** [comment]
Doing so gives me a

```
cannot find module "Author"
```

**Comment by jeffijoe** [comment]
@HamletDRC additionally, this library itself seems to use relative paths:

![image](https://cloud.githubusercontent.com/assets/2505474/16611783/c5151658-4363-11e6-9fb1-32fb8a8d388a.png)

**Comment by HamletDRC** [comment]
If your code is just loading files from disk then I think you will need to use relative imports. You can only avoid them if you are using a module system like requirejs or webpack. We use both at MSE and the rule works well for us. tslint just loads the files off disk so there is no way to use the rule on this project. 

It sounds to me like you're deployment requires you to use relative paths.

**Comment by jeffijoe** [comment]
I just haven't come across a configuration where relative paths were never used, unless you had only a single file of custom code and only imported npm modules :smiley:

**Comment by HamletDRC** [comment]
We never use relative paths on Microsoft Social Engagement (100kloc). We use requirejs or webpack for our deployments. Each class/module is in a single file so we have hundreds of TypeScript files. r.js or webpack takes cares of all the bundling for us.

**Comment by jeffijoe** [comment]
So every file has to have a unique name?

**Comment by HamletDRC** [comment]
No. My current import list looks like this: 

    import chai = require('chai');
    import PostDomElement = require('post-interactions/tests/PostDomElement');
    import PostsController = require('post-interactions/models/PostsController');
    import PostsWidgetTestHelper = require('tests/post-interactions/components/post-list/PostsWidgetTestHelper');
    import React = require('react');

react and chai are in ./lib and the others are files on disk in the subdirectories specified. 
Locally we run grunt server and serve individual files through http and in deployments (DEV, Testing, etc) we concatenate everything into a few files and have requirejs load everything correctly.

**Comment by HamletDRC** [comment]
Are you writing a node application? Honestly, I only have experience with browser based deployments.

**Comment by jeffijoe** [comment]
Ah, so all your paths are relative to the source root?

Yes, I am writing a node app. :)

**Comment by HamletDRC** [comment]
I would say all paths are not relative to the source root... in my example that is true but the project is split into several sub-projects such as "common" and "widgets". Other imports navigate up and over directories. But we had to configure the build to allow this and generate d.ts files for local sub-projects. 

Anyway, I suspect that you are fine to use relative paths. I personally think it is easier without but a typical node app might well use them.
